src/lib.rs

```diff
@@ -20,7 +20,7 @@ fn merge(objs: &PyTuple) -> PyResult<PyObject> {
         }
 
         let current = py_object_to_json(obj)?;
-        merged = merge_json_objects(&merged, &current);
+        merged = merge_json_objects(&merged, &current)?;
```

Comment:
`merge_json_objects` can return a `PyErr` now, if it gets a `-history` key with a non-list value – which adds a bunch of `?` and `.unwrap()` and `Ok()` to the code.